datamol/utils/jobs.py

```diff
@@ -251,3 +255,82 @@ def parallelized(
         **job_kwargs,
     )
     return runner(fn, inputs_list, arg_type=arg_type)
+
+
+def parallelized_with_batches(
```

Comment:
this vs:

`dm.parallelized(fn, inputs, n_jobs=-1, batch_size=N)` (the batch_size arg is forward automatically to joblib, although not documented)

Reply:
curious about speed benchmark

Reply:
Yup good question but the difference is important here.

`batch_size` in joblib will simply schedule batches of inputs to the workers but `_fn` still works on a single input element. I did benchmark `batch_size` a while ago for circus and it does not change that much. Under the hood joblib automatically tunes it to lower the overhead so most of the time setting it to `None` is the best thing to do.

`batch_size` in `parallelized_with_batches` is different since here `_fn` works on a batch of inputs and not a single element anymore. So the speedup will be obvious if your `_fn` needs to do some init work no matter the number of elements.

Reply:
Any idea to improve the docstring in order to make things more clear is welcome.

Reply:
>  if your _fn needs to do some init work no matter the number of elements.

beside that, they will likely be no gain (neither will take longer).

Reply:
I think the likely huge advantage would be around operation that can be vectorized (example, tensor/numpy array or dataframe)

Reply:
> I think the likely huge advantage would be around operation that can be vectorized (example, tensor/numpy array or dataframe)

Yup exactly this (but also anything that require a costly init logic, see our `catalog_filter` for example).